src/sentry/static/sentry/app/stores/alertStore.jsx

```diff
@@ -9,34 +11,61 @@ const AlertStore = Reflux.createStore({
     this.count = 0;
   },
 
-  onAddAlert(message, type, expireAfter, url) {
-    // intentionally recreate array via concat because of Reflux
-    // "bug" where React components are given same reference to tracked
-    // data objects, and don't *see* that values have changed
-    let alertId = this.count++;
+  onAddAlert(alert) {
+    if (defined(alert.id)) {
+      let expirations = getItem('alerts:muted');
+      if (defined(expirations)) {
+        expirations = JSON.parse(expirations);
 
-    this.alerts = this.alerts.concat([{
-      id: alertId,
-      message: message,
-      type: type,
-      url: url
-    }]);
+        // Remove any objects that have passed their mute duration.
+        let now = Math.floor(new Date() / 1000);
+        for (let key in expirations) {
+          if (expirations.hasOwnProperty(key) && expirations[key] < now) {
+            delete expirations[key];
+          }
+        }
+        setItem('alerts:muted', JSON.stringify(expirations));
```

Comment:
This can write back an empty object, I can change it if we'd rather just remove it entirely.